### PR TITLE
test: add instrumentation test to component Text

### PR DESCRIPTION
**Pull request by FredericoNoce**
### Related Issues

https://github.com/ZupIT/beagle/issues/1013

### Description and Example

Creates automated tests for text with and without expressions, including:

text
textColor
textAlignment

- [x] I am willing to follow-up on review comments in a timely manner.

[PR Guide]: https://github.com/ZupIT/beagle/blob/master/doc/contributing/pull_requests.md

### CODE REVIEW

**Comment by botbeaglezup** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">The title of PR not follow the rules of: https://www.conventionalcommits.org/en/v1.0.0/</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Big PR</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: backend/automated-tests/src/main/kotlin/br/com/zup/beagle/automatedtests/builders/TextScreenBuilder.kt [hunk 1]
```diff
@@ -58,33 +67,42 @@ object TextScreenBuilder {
...
-                beagleText(text = "@{text.textBackground}", appearanceColor = STEEL_BLUE),
-                beagleText(
-                    text = "@{text.textAll}",
-                    styleId = SCREEN_TEXT_STYLE,
-                    appearanceColor = STEEL_BLUE
-                )
+                beagleText(text = "hello world"),
+                beagleText(text = expressionOf("@{context.text}")),
+                beagleText(text = "hello world with textColor", textColor = LIGHT_GREY),
+                beagleText(text = expressionOf("@{context.textExpressionColor}"), textColor = expressionOf("@{context.color}")),
```
Comment (botbeaglezup) [inline backend/automated-tests/src/main/kotlin/br/com/zup/beagle/automatedtests/builders/TextScreenBuilder.kt:73]: <table data-meta="generated_by_backend">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/d5ff656522c44b245029d430d2c88663c1820d1d/backend/automated-tests/src/main/kotlin/br/com/zup/beagle/automatedtests/builders/TextScreenBuilder.kt#L73"></span>Line detected that is longer than the defined maximum line length in the code style.</td>
    </tr>
  </tbody>
</table>

File: backend/automated-tests/src/main/kotlin/br/com/zup/beagle/automatedtests/builders/TextScreenBuilder.kt [hunk 2]
```diff
@@ -58,33 +67,42 @@ object TextScreenBuilder {
...
+                beagleText(text = "hello world with textAlignment LEFT", alignment = TextAlignment.LEFT),
+                beagleText(text = "hello world with textAlignment CENTER", alignment = TextAlignment.CENTER),
+                beagleText(text = "hello world with textAlignment RIGHT", alignment = TextAlignment.RIGHT),
+                beagleText(text = expressionOf("@{context.textExpressionAlignmentLeft}"), alignment = expressionOf("@{context.alignmentLeft}")),
```
Comment (botbeaglezup) [inline backend/automated-tests/src/main/kotlin/br/com/zup/beagle/automatedtests/builders/TextScreenBuilder.kt:77]: <table data-meta="generated_by_backend">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/d5ff656522c44b245029d430d2c88663c1820d1d/backend/automated-tests/src/main/kotlin/br/com/zup/beagle/automatedtests/builders/TextScreenBuilder.kt#L77"></span>Line detected that is longer than the defined maximum line length in the code style.</td>
    </tr>
  </tbody>
</table>

File: backend/automated-tests/src/main/kotlin/br/com/zup/beagle/automatedtests/builders/TextScreenBuilder.kt [hunk 3]
```diff
@@ -58,33 +67,42 @@ object TextScreenBuilder {
...
+                beagleText(text = expressionOf("@{context.textExpressionAlignmentCenter}"), alignment = expressionOf("@{context.alignmentCenter}")),
```
Comment (botbeaglezup) [inline backend/automated-tests/src/main/kotlin/br/com/zup/beagle/automatedtests/builders/TextScreenBuilder.kt:78]: <table data-meta="generated_by_backend">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/d5ff656522c44b245029d430d2c88663c1820d1d/backend/automated-tests/src/main/kotlin/br/com/zup/beagle/automatedtests/builders/TextScreenBuilder.kt#L78"></span>Line detected that is longer than the defined maximum line length in the code style.</td>
    </tr>
  </tbody>
</table>

File: backend/automated-tests/src/main/kotlin/br/com/zup/beagle/automatedtests/builders/TextScreenBuilder.kt [hunk 4]
```diff
@@ -58,33 +67,42 @@ object TextScreenBuilder {
...
+                beagleText(text = expressionOf("@{context.textExpressionAlignmentRight}"), alignment = expressionOf("@{context.alignmentRight}"))
```
Comment (botbeaglezup) [inline backend/automated-tests/src/main/kotlin/br/com/zup/beagle/automatedtests/builders/TextScreenBuilder.kt:79]: <table data-meta="generated_by_backend">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/d5ff656522c44b245029d430d2c88663c1820d1d/backend/automated-tests/src/main/kotlin/br/com/zup/beagle/automatedtests/builders/TextScreenBuilder.kt#L79"></span>Line detected that is longer than the defined maximum line length in the code style.</td>
    </tr>
  </tbody>
</table>

**Comment by FredericoNoce** [comment]
Waiting for https://github.com/ZupIT/beagle/issues/958 to merge.
Tests 8, 9 and 10 will work after merging above.

File: backend/automated-tests/src/main/kotlin/br/com/zup/beagle/automatedtests/builders/TextScreenBuilder.kt [hunk 5]
```diff
@@ -16,14 +16,14 @@
...
 
-import br.com.zup.beagle.automatedtests.constants.SCREEN_TEXT_STYLE
-import br.com.zup.beagle.automatedtests.constants.STEEL_BLUE
+import br.com.zup.beagle.automatedtests.constants.LIGHT_GREY
 import br.com.zup.beagle.core.Style
 import br.com.zup.beagle.ext.applyStyle
 import br.com.zup.beagle.ext.unitReal
 import br.com.zup.beagle.widget.action.Alert
-import br.com.zup.beagle.widget.context.ContextData
+import br.com.zup.beagle.widget.context.*
```
Comment (botbeaglezup) [inline backend/automated-tests/src/main/kotlin/br/com/zup/beagle/automatedtests/builders/TextScreenBuilder.kt:24]: <table data-meta="generated_by_backend">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/016509642b2f4ecf22741a7ca5d4f47e2438c156/backend/automated-tests/src/main/kotlin/br/com/zup/beagle/automatedtests/builders/TextScreenBuilder.kt#L24"></span>br.com.zup.beagle.widget.context.* is a wildcard import. Replace it with fully qualified imports.</td>
    </tr>
  </tbody>
</table>

**Review by carlossteinzup — APPROVED** [review]

**Review by carlossteinzup — APPROVED** [review]

**Comment by theffc** [comment]
It seems like this PR should not be named `feat:` since it's not adding some new feature for users, it instead adds an internal test. So, I would suggest we use some other prefix in these PRs that only change integration tests, maybe `tests: ` would be good.
For other information on this convention of prefixing commit messages, check [this](https://www.conventionalcommits.org/en/v1.0.0/)

**Review by viniciusguardieirozup — APPROVED** [review]

**Review by uziasferreirazup — APPROVED** [review]

**Pull request merged**